### PR TITLE
feat(variables): add toggle enable_interpolation_in_file

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-1257

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview b1cc6d03-1b9e-4536-b5ee-dc186c55218c | [dev](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/b1cc6d03-1b9e-4536-b5ee-dc186c55218c/services/general) |
 /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Help yourself at 2AM by putting some metrics to help observability

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx [hunk 1]
```diff
@@ -221,6 +224,8 @@ export function CreateUpdateVariableModal(props: CreateUpdateVariableModalProps)
         'Variable are used at build/run time. Secrets are special variables, their value can only be accessed by the application.'
     )
 
+  console.log(type)
```
Comment (ctjhoa) [inline libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx:227]: to clean

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx]: https://github.com/Qovery/console/pull/1576/commits/2ab42bfc3b848967b82c1ba75d68993d41a660aa

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx [hunk 2]
```diff
@@ -87,6 +88,7 @@ export function CreateUpdateVariableModal(props: CreateUpdateVariableModalProps)
       value: variable?.value ?? '',
       isSecret: variable?.is_secret,
       description: variable?.description,
+      enable_interpolation_in_file: variable?.enable_interpolation_in_file,
```
Comment (ctjhoa) [inline libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx:91]: You miss one point of the spec
2024-08-20-180914_600x239_scrot

```suggestion
      enable_interpolation_in_file: variable?.enable_interpolation_in_file ?? true,
```

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx]: Fixed, thanks https://github.com/Qovery/console/pull/1576/commits/d3b06aecf7173f3033bd39614c68bfb71d4a21a0

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**